### PR TITLE
Ports: Fix and update zstd port

### LINKED ISSUE #10066: Ports: zstd fails to build on MacOS
zstd has some cross-compile issues in MacOS currently. From the ports build output:

```
Building zstd...
+ make -j8
compiling dynamic library 1.4.4
i686-pc-serenity-gcc: error: unrecognized command-line option '-install_name'
i686-pc-serenity-gcc: error: unrecognized command-line option '-compatibility_version'
i686-pc-serenity-gcc: error: unrecognized command-line option '-current_version'
make[1]: *** [libzstd.1.4.4.dylib] Error 1
make: *** [lib-release] Error 2
make: *** Waiting for unfinished jobs....
make[1]: Nothing to be done for `zstd-release'.
cp programs/zstd .
```

From discussion on Discord:

```
<Andrew_Kaster> here's the problem https://github.com/facebook/zstd/blob/71076b7a014e70c74153b24b6a2ab106adfd3b4c/lib/Makefile#L207-L219
<Andrew_Kaster> need to make it not think that we are compiling for darwin
<Andrew_Kaster> hmm. looks like this Makefile isn't particularly friendly to cross compiling
<Andrew_Kaster> I think the right fix is to use the CMake build so we can pass the toolchain file
```

Removing part of the `ifeq ($(UNAME), Darwin)` code seems to be a workaround for this, for now.

### PR DESCRIPTION
**Pull request by EWouters**
Updated the version of zstd to 1.5.0 and fixed the bug that was present on Mac when installing the zstd port.

Fixes #10066

PS: this is my first PR here, please tell me if I'm doing something wrong 😬

### CODE REVIEW

File: Ports/zstd/patches/fix_soname_flags.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- zstd-1.5.0/lib/Makefile.orig	2021-11-01 15:49:13.000000000 +0100
++++ zstd-1.5.0/lib/Makefile	2021-11-01 15:49:16.000000000 +0100
+@@ -210,7 +210,7 @@
+   SHARED_EXT = dylib
```
Comment (timschumi) [inline Ports/zstd/patches/fix_soname_flags.patch:4]: Although I can't spot anything with dylib ending up in the final file system, I feel like we should just remove the whole Darwin-specific case for clarity's sake, and just rely on what the library would do on a non-Apple system.

**Review by timschumi — COMMENTED** [review]
Small nitpick in the comments, looks reasonable otherwise (and still builds with i686 and GCC on a Linux host).

File: Ports/zstd/patches/fix_soname_flags.patch (hunk 1 above)
Comment (EWouters) [reply Ports/zstd/patches/fix_soname_flags.patch:4]: That sounds like it makes more sense than what I did. Do you know how to test this?

Comment (timschumi) [reply Ports/zstd/patches/fix_soname_flags.patch:4]: I think the only way that you can really test this is by building this on macOS (which I just realized I didn't, so of course I haven't seen any dylib files) and manually checking for *.dylib files in the resulting disk image.

**Comment by EWouters** [comment]
Fixed the PR and commit titles to make the linter happy.

File: Ports/zstd/patches/fix_soname_flags.patch (hunk 1 above)
Comment (EWouters) [reply Ports/zstd/patches/fix_soname_flags.patch]: I took your suggestion, so it will now produce a .so library.

**Comment by timschumi** [comment]
Please don't ping users in commit messages, it generates a notification every time someone pushes that commit to any repository ;-)

**Comment by EWouters** [comment]
> Please don't ping users in commit messages, it generates a notification every time someone pushes that commit to any repository ;-)

Sorry, I'll keep that in mind.

**Comment by awesomekling** [comment]
> > Please don't ping users in commit messages, it generates a notification every time someone pushes that commit to any repository ;-)
> 
> Sorry, I'll keep that in mind.

This PR still has a commit that pings a user.

**Pull request merged**